src/org/mozilla/javascript/NativeString.java

```diff
@@ -1106,6 +1110,47 @@ protected int findPrototypeId(Symbol k) {
         return 0;
     }
 
+    /**
+     *
+     *
+     * <h1>String.raw (template, ...substitutions)</h1>
+     *
+     * <p>22.1.2.4 String.raw [Draft ECMA-262 / April 28, 2021]
+     */
+    private static CharSequence js_raw(Context cx, Scriptable scope, Object[] args) {
+        /* step 1-2 */
+        Object arg0 = args.length > 0 ? args[0] : Undefined.instance;
+        Scriptable cooked = ScriptRuntime.toObject(cx, scope, arg0);
+        /* step 3 */
+        Object rawValue = ScriptRuntime.getObjectProp(cooked, "raw", cx);
+        Scriptable raw = ScriptRuntime.toObject(cx, scope, rawValue);
+        /* step 4-5 */
+        long rawLength = NativeArray.getLengthProperty(cx, raw);
+        if (rawLength > Integer.MAX_VALUE) {
+            throw ScriptRuntime.rangeError("raw.length > " + Integer.toString(Integer.MAX_VALUE));
+        }
+        int literalSegments = (int) rawLength;
+        if (literalSegments <= 0) return "";
+        /* step 6-7 */
+        StringBuilder elements = new StringBuilder();
+        int nextIndex = 0;
+        for (; ; ) {
+            /* step 8 a-i */
+            Object next;
+            next = ScriptRuntime.getObjectIndex(raw, nextIndex, cx);
```

Comment:
This also changed from the original PR to use `ScriptRuntime.getObjectIndex` instead of `Scriptable.get(int Scriptable)`, which searches up the prototype chain and returns undefined when not found, rather than only searching the current object, and returning Scriptable.NOT_FOUND.